core/rendered_connection.ts

```diff
@@ -559,21 +555,6 @@ export class RenderedConnection
     childBlock.updateDisabled();
     childBlock.queueRender();
 
-    // If either block being connected was selected, visually un- and reselect
-    // it. This has the effect of moving the selection path to the end of the
-    // list of child nodes in the DOM. Since SVG z-order is determined by node
-    // order in the DOM, this works around an issue where the selection outline
-    // path could be partially obscured by a new block inserted after it in the
-    // DOM.
-    const selection = common.getSelected();
-    const selectedBlock =
-      (selection === parentBlock && parentBlock) ||
-      (selection === childBlock && childBlock);
-    if (selectedBlock) {
-      selectedBlock.removeSelect();
-      selectedBlock.addSelect();
-    }
```

Comment:
Check whatever cases this can occur to see if it still works as expected.

Reply:
Note that this testing will include verifying https://github.com/google/blockly/pull/9004#discussion_r2078738887 as well to keep things in one location for `RenderedConnection`.

Reply:
I've dug back to https://github.com/google/blockly/commit/02b2c219ebab921b3c725f1e8ac20ebc80e29202 as the original introducing of the selection correction logic. I'm not sure what the original bug's repro steps were, but this fix is very old and predates gestures plus bring-to-front logic (at least in its current form). I'm now suspecting that the selection bits for bumping have been made obsolete with other bring-to-front mechanisms such that this hasn't been needed for some time. No amount of bumping seems to get in the way of the selected block or the selection itself (both always seem to stay above the bumped block), so I do think it's safe to remove this logic now.

Reply:
Going back to `connect_`, I think I know what's happening here and verifying would be tricky, but I think this fine to logic out based on the observation that everything is working fine without the selection bits in.

With the latest changes, the data flow roughly looks like this:
- User sends pointer event to click on a block.
- Gestures mark that block as focused.
- The block, upon receiving focus, auto select themselves (and if a previous block was focused, it removes its selection).
- The process of removing/adding a selection bumps it back to the top in the same way as the old logic here did.

Thus, this shouldn't be needed since the various efforts to focus the block being moved should generally always result with its selection appearing on top.